app/code/core/Mage/Backup/Model/Resource/Db.php

```diff
@@ -87,7 +87,9 @@ public function getTables()
      */
     public function getTableDropSql($tableName)
     {
-        return Mage::getResourceHelper('backup')->getTableDropSql($tableName);
+        /** @var Mage_Backup_Model_Resource_Helper_Mysql4 $helper */
```

Comment:
1) we should not use mysql4 classes, they are deprecated, arent they?
2) this can be fixed in the phpstan plugin, so phpstan knows what types getResourceHelper factory returns without typehints (similar to how other factories are handled)

Reply:
1. resource helpers are not deprecated (e.g. `class Mage_Core_Model_Resource_Helper_Mysql4 extends Mage_Core_Model_Resource_Helper_Abstract`)
2. change to draft and check your phpstan plugin :+1:

Reply:
@tmotyl checked your plug-in, but it seems more difficult in this case b/c there some more functions called that have to be added.

I'll work on this later (still useful for extensions), but this PR would solve the issues for native phpstan ...

Reply:
ok.
about mysql4 -> I thought that using classes with mysql4 in name is discouraged, so we should not use them in typehints

Reply:
The mysql4 classes have been deprecated for models/resource-models. The resource-helper is another story ...

Class description 
```
 * Resource helper class for MySql Varien DB Adapter
```

If i have not did not wrong, the mysql4-name is related to ...

https://github.com/OpenMage/magento-lts/blob/76bae0874d1a113684584720e097000c841b4d02/app/etc/config.xml#L34